src/main/resources/META-INF/plugin.xml

```diff
@@ -188,67 +188,68 @@
         <applicationService serviceImplementation="org.sonarlint.intellij.config.global.SonarLintGlobalSettingsStore"/>
+
+        <!-- Cannot be moved to service annotation because of 'serviceInterface' / 'testServiceImplementation' -->
         <applicationService serviceInterface="org.sonarlint.intellij.core.EngineManager"
                             serviceImplementation="org.sonarlint.intellij.core.DefaultEngineManager"
                             testServiceImplementation="org.sonarlint.intellij.core.TestEngineManager"/>
+
         <applicationService serviceImplementation="org.sonarlint.intellij.SonarLintPlugin"/>
+
+        <!-- Cannot be moved to service annotation because of 'serviceInterface' / 'testServiceImplementation' -->
         <applicationService serviceInterface="org.sonarlint.intellij.util.GlobalLogOutput"
                             serviceImplementation="org.sonarlint.intellij.util.GlobalLogOutputImpl"
                             testServiceImplementation="org.sonarlint.intellij.util.GlobalLogOutputTestImpl"/>
+
         <applicationService serviceImplementation="org.sonarlint.intellij.util.SonarLintActions"/>
+
+        <!-- Cannot be moved to service annotation because of 'serviceInterface' -->
         <applicationService serviceInterface="org.sonarlint.intellij.telemetry.SonarLintTelemetry"
                             serviceImplementation="org.sonarlint.intellij.telemetry.SonarLintTelemetryImpl" />
-        <applicationService serviceImplementation="org.sonarlint.intellij.core.ServerNotificationsService"/>
-        <applicationService serviceImplementation="org.sonarlint.intellij.core.NodeJsManager"/>
-        <applicationService serviceImplementation="org.sonarlint.intellij.fs.ModuleFileEventsNotifier"/>
-        <applicationService serviceImplementation="org.sonarlint.intellij.module.ModulesRegistry"/>
+
         <applicationService serviceImplementation="org.sonarlint.intellij.core.BackendService"/>
+
+        <!-- Cannot be moved to service annotation because of 'serviceInterface' / 'testServiceImplementation' -->
         <applicationService serviceInterface="org.sonarlint.intellij.util.ThreadPoolExecutor"
                             serviceImplementation="org.sonarlint.intellij.util.ApplicationThreadPoolExecutor"
                             testServiceImplementation="org.sonarlint.intellij.util.SynchronousThreadPoolExecutor"/>
 
-        <projectService serviceImplementation="org.sonarlint.intellij.core.ProjectBindingManager"/>
+        <!-- Cannot be moved to service annotation because of 'serviceInterface' -->
         <projectService serviceInterface="org.sonarlint.intellij.common.vcs.VcsService"
                         serviceImplementation="org.sonarlint.intellij.vcs.DefaultVcsService"/>
+
+        <!-- Cannot be moved to service annotation because of persisting project state -->
         <projectService serviceImplementation="org.sonarlint.intellij.config.project.SonarLintProjectSettingsStore"/>
```

Comment:
I saw classes in the platform that use both `@Service` and `@State` for persistent storage, so I think it could work

Reply:
I'll try that on Monday.

Reply:
Done, see the question regarding the the application service in the slack channel for the last state